src/wasm.h

```diff
@@ -565,10 +568,35 @@ class Expression {
     assert(int(_id) == int(T::SpecificId));
     return (const T*)this;
   }
+
+  bool isConstExpression() const {
+    switch (_id) {
+      case ConstId:
+      case RefNullId:
+      case RefFuncId:
+        return true;
+      default:
+        return false;
+    }
+  }
+
```

Comment:
Unless the functions are in this header file, I don't think it will increase the object size. So we can only leave the declarations here and put the definitions in wasm.cpp. Or you don't like the functions are in this `Expression` class, whether the definitions are in this file or not? I think we have three options:
1. Leave the declaration here and put the definition in wasm.cpp
2. Take the function out of `Expression` but still leave the declaration in this file and put the definition in wasm.cpp
3. Move both the declaration and definition to another file (where?)

For 2 there's a downside of cluttering `wasm` namespace, which is effectively the global namespace here. I think 1 should be fine, because if we put the definition to the wasm.cpp, it's not gonna increase the object file size after all. For `canInitializeGlobal`, I don't mind moving `canInitializeGlobal` to `ir/global-utils.h`.

Reply:
We talked in person, and @azakai's concern was not about object file size but he wanted to keep `wasm.h` as minimal as possible, so that it only contains the basic structure of wasm module and people can scan it to understand the structure quickly. I moved those functions to `ir/properties.h`.

We also talked about whether it would be better to use `Const` class for not only `[type].const` instructions but also for `ref.null` and `ref.func` instructions, now that they are constants too. That's a rather big change because `Const` is used everywhere, so we can think about it for a follow-up patch.